test/services/CheckoutServiceSpec.scala

```diff
@@ -5,24 +5,28 @@ import com.gu.membership.salesforce.{BasicMember, MemberId}
 import com.gu.membership.zuora.soap.Zuora.{Authentication, SubscribeResult}
 import model.{PaymentData, PersonalData, SubscriptionData}
 import org.scalatest.FreeSpec
-import services.CheckoutService.CheckoutResult
+import org.scalatest.concurrent.{Futures, ScalaFutures}
+import org.scalatest.time.{Millis, Seconds, Span}
 import utils.ScheduledTask
 import utils.TestPersonalData._
 
 import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.duration._
-import scala.concurrent.{Await, Future}
+import scala.concurrent.Future
 
-class CheckoutServiceSpec extends FreeSpec {
+class CheckoutServiceSpec extends FreeSpec with Futures with ScalaFutures {
+  implicit override val patienceConfig =
+    PatienceConfig(timeout = scaled(Span(10, Seconds)), interval = scaled(Span(50, Millis)))
 
-  def makeIdentityService(onUserUpdate: => Unit = {}): IdentityService=
+  class UpdateFlag(var updated: Boolean = false)
+
+  def makeIdentityService(updateFlag: UpdateFlag): IdentityService=
```

Comment:
The `updateFlag` argument could probably have a default; this would make the tests a bit more dry